src/circuits/Network.stanza

```diff
@@ -0,0 +1,493 @@
+#use-added-syntax(jitx)
+doc: \<DOC>
+@brief Lump Circuit Networks
+
+The concept of this code is to provide a means for defining a lumped
+circuit network tree. The idea is to provide a "DSL" for defining
+components that are used together to form a circuit network.
+
+In this graph there are Elements and Nodes.
+
+* Elements are the edges of the graph - the resistors, capacitors, etc.
+* Nodes are the pins, ground, nets, etc of the circuit
+
+We can think of the circuit network as a graph. We have the following
+ operations to this graph for 2-pin elements
+
+ 1.  Series (+) - string multiple elements in series
+ 2.  Parallel (|) - put multiple elements in parallel
+ 3.  Tap - For a particular node in the circuit graph, connect this node to
+     an external port on the resultant network instantiable (pcb-module). The
+     components are still connected on both sides in the network.
+ 4.  ShuntTo - Connect a given node to a port through an element.
+     This allows connecting a component that escapes the network. For example,
+     this might be used connect a particular component or sub-network to ground.
+ 5.  Invert (~) - Instead of connecting 1 -> 2 , connect 2 -> 1. reverses the
+       orientation of the component. Useful for polarizing caps, diodes, etc.
+
+
+<DOC>
+defpackage jsl/circuits/Network :
+  import core
+  import collections
+  import jitx
+  import jitx/commands
+
+  import jsl/design/introspection
+  import jsl/errors
+  import jsl/ensure
+
+doc: \<DOC>
+Abstract Network Type
+
+The circuit tree is composed of `ANT` instances that form
+a sequence of operations for the tree.
+<DOC>
+public deftype ANT
+doc: \<DOC>
+Each `ANT` instance contains a recursive set of ANT elements
+
+The elements of each ANT instance are interpreted by that node.
+Some may be empty, others may have a single instance, and others
+may have 1 or more instances.
+<DOC>
+public defmulti elements (a:ANT) -> IndexedCollection<ANT>
+
+
+val DEF_ELEM_NAME = `elem
+val DEF_ELEM_PUBLIC = false
+doc: \<DOC>
+Circuit Element for a Circuit Network
+
+This type typically includes an instantiable that will be
+instantiated in the resultant network definition.
+<DOC>
+public defstruct CircuitElement <: ANT:
+  doc: \<DOC>
+  Elements of this node - This Empty by default for `CircuitElement`
```

Comment:
I am not sure that I understand this sentence.

Reply:
Oops